machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,89 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfig is the Schema for the nutanixmachineproviderconfigs API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// clusterReference is to identify the PE/cluster in which the Machine's VM will be created
+	// +kubebuilder:validation:Required
+	ClusterReference *NutanixReference `json:"clusterReference"`
```

Comment:
What is `PE`? Can we spell that out please.
Are there any restrictions on this reference? Does it for example need to match any other values? Eg should it be matching the infrastructure object?

We don't need to include `Reference` in the name here, or any of the other references 

Are both UUID and Name suitable ways to look up a cluster?

I don't think this needs to be a pointer, can we make it a reference directly please
```suggestion
	// cluster is used to identify the P... E... /cluster in which the Machine's VM will be created.
	// The cluster UUID can be found ...
	// Alternatively you can identify the cluster by name.
	// +kubebuilder:validation:Required
	Cluster NutanixReference `json:"clusterReference"`
```

Reply:
Addressed